### ISSUE TITLE
Random CV - how does it work?

### ISSUE THREAD

**Issue by statslover123** [brandon-mosqueda/SKM#2]
Hello,

I am using bayesian optimization with random as my CV option with .2 missing.

Does 15 tune_folds_number mean there are 15 seperate traint / test splits with .8 and .2 respectively?

**Comment by brandon-mosqueda** [comment]
Hello,

that's correct, in each fold the .8 of data is used for training and .2 for testing, these proportions are the same in all folds but the observations are selected at random at each fold, so they are diferent.

Keep also in mind that _tune_folds_number_ is the parameter to specify with how many folds you want to evaluate **each  hyperparameters combination** generated by bayesian optimization algorithm. With parameters _tune_bayes_samples_number_  and _tune_bayes_iterations_number_, you select the initial number of samples generated at random for the hyperparameters and the number of samples you want to generate and evaluate with the optimization algorithm, respectively.

With bayesian optimization, at the end you are going to evaluate these many folds:

tune_folds_number x (tune_bayes_samples_number + tune_bayes_iterations_number) + 1

that is because you are going to evaluate _tune_bayes_samples_number_ + _tune_bayes_iterations_number_ different hyperparameters combinations _tune_folds_number_ times each one and the final model fitted with the whole dataset and the best combination of hyperparameters (+1).

**Comment by statslover123** [comment]
thank you! 

I am not sure I understand the difference between tune_bayes_sampler_number vs tunes_bayes_iterations_number...
Is there a good rule of thumb to follow for these two values?

**Comment by brandon-mosqueda** [comment]
The first one (tune_bayes_sampler_number) defines how many random hyperparameters combinations you want to evaluate at start, with these first hyperparameters and the loss values produced, the bayesian optimization algorithm can propose the next values to be evaluated based on their distributions and wanting to reduce the loss value.
With tunes_bayes_iterations_number you specify how many of these optimization iterations want to evaluate.

Usually you want tunes_bayes_iterations_number be greater because the first proposed values are generated at random so you do not expect great performance with them, but in the in each next optimization iteration is expected (in theory) that new parameters improves the performance.

But more iterations are also more expensive because of inner cross validation, so find a trade-off that works for your problem. By default it uses 10 and 10, but I think 5-10 or 5-15 are okey too.

**Comment by statslover123** [comment]
I understand now thank you!
